### PR TITLE
Add option to disable time elapsed on discord activity

### LINKED ISSUE #5282: Option to disable time elapsed on discord activity
On the discord configuration would it be possible to put a Show Time Elapsed toggle, so people clicking on your discord profile would not be able to see how long you have been playing, only what activities you are currently doing.

### PR DESCRIPTION
**Pull request by MSAskew**
Updated plugin to allow disabling of elapsed time. EventWithTime start property now nullable and will be set depending upon config checkbox.
Also fixed a comment that was a copy paste error.

Closes #5282

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordConfig.java [hunk 1]
```diff
@@ -31,11 +31,22 @@
 @ConfigGroup("discord")
 public interface DiscordConfig extends Config
 {
+	@ConfigItem(
+		keyName = "showElapsedTime",
+		name = "Show elapsed time",
+		description = "Configures if the elapsed time of your activity should be shown.",
+		position = 1
+	)
+	default boolean showElapsedTime()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordConfig.java]: You should probably invert this and make it hideElapsedTime

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordConfig.java]: Also it should go below action timeout

File: runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java [hunk 2]
```diff
@@ -45,7 +45,7 @@
 	private class EventWithTime
 	{
 		private final DiscordGameEventType type;
-		private final Instant start;
+		private Instant start;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java]: this should still stay final

File: runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java [hunk 3]
```diff
@@ -87,7 +87,13 @@ void triggerEvent(final DiscordGameEventType eventType)
 		}
 		else
 		{
-			event = new EventWithTime(eventType, Instant.now());
+			event = new EventWithTime(eventType);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java]: just send null to constructor if hideElapsedTime is true

**Comment by MSAskew** [comment]
Updates made following @deathbeam's feedback.

File: runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java [hunk 4]
```diff
@@ -87,7 +87,17 @@ void triggerEvent(final DiscordGameEventType eventType)
 		}
 		else
 		{
-			event = new EventWithTime(eventType, Instant.now());
+			if (config.hideElapsedTime())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java]: `event = new EventWithTime(eventType, config.hideElapsedTime() ? null : Instant.now());` maybe?

**Pull request merged**